core/src/main/java/io/questdb/std/str/Int3Sort.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.std.str;
+
+public class Int3Sort {
+
+    private static final int N = 3;
+
+    public static void quickSort(int[] array, int low, int high) {
+        if (low + 1 < high) {
```

Comment:
* Assert size of array.
* Assert bounds of low and high.
* Test said asserts.

Reply:
I don't see much of a point.

1. You'll get IndexOutOfRangeException without the assertion, what else would the assertion add?
2. Same
3. Same

Reply:
You're making the assumption that will be the behaviour, and it may be, but it's not entirely clear to me this is the case in all circumstances.

1. An assert will give allow you to spot an illegal use immediately with clarity, an exception could be either an illegal use or a bug in the sorting impl.
2. Again, clarity of errors -- assuming they're caught. What I meant here avoid invalidly-aligned indices. I'm assuming that the integers here need to be `% N == 0`, no?
3. Asserts are disabled in production code, no? It shouldn't cause bad behaviour even if they're off.

Reply:
I ended up adding 1 assertion. I moved the visibility of the method to be package interval so that it's only accessible using `IntegerList.sortGroups()` method, it has a few more assertions.